models/VendorHours.tsx

```diff
@@ -1,42 +1,91 @@
-
 type DayOfWeek = "Monday" | "Tuesday" | "Wednesday" | "Thursday" | "Friday" | "Saturday" | "Sunday";
 
+type TimeRange = {
+  open: string;
+  close: string;
+};
+
 type VendorHours = {
-    [key in DayOfWeek]: {
-      open: string;
-      close: string;
-    };
+  [key in DayOfWeek]: TimeRange[];
+};
+
+const formatTimeRange = (timeRange: TimeRange): string => {
+  return `${timeRange.open} - ${timeRange.close}`;
+};
+
+const getVendorHoursForDayString = (vendorHours: VendorHours, day: DayOfWeek, includeDay: boolean = false): string => {
+  let hoursString = '';
+    const timeRanges = vendorHours[day];
+
+  if (timeRanges.length > 0) {
+    var dayHoursString = '';
+    if(includeDay){
+      dayHoursString += `${day}: `;
+    }
+    dayHoursString += timeRanges.map(formatTimeRange).join(', ');
+    hoursString += dayHoursString;
+  }
+
+  return hoursString.trim(); // Trim to remove any trailing newline
+};
+
+const vendorHoursToString = (vendorHours: VendorHours): string => {
+  let hoursString = '';
+
+  for (const day in vendorHours) {
+    hoursString += getVendorHoursForDayString(vendorHours, day as DayOfWeek);
+  }
+
+  return hoursString;
+}
+
+const isVendorOpenHelper = (vendorHours: VendorHours, day: DayOfWeek, currentTime: string): boolean => {
+  const todayHours = vendorHours[day];
+
+  if (!todayHours) {
+    return false;
+  }
+
+  for (const interval of todayHours) {
+    const { open, close } = interval;
+    if (currentTime >= open && currentTime <= close) {
+      return true; 
+    }
+  }
+
+  return false;
+};
+
+const isVendorCurrentlyOpen = (vendorHours: VendorHours): boolean => {
```

Comment:
This method currently works on UTC Time, doesn't show it correctly for PST needs fix